tests/utest/test_api.py

```diff
@@ -29,9 +29,41 @@ def test_run_check_in_memory(self):
         in_memory = "*** Settings ***\n\n"
         ast_model = get_model(in_memory)
         issues = robocop_runner.run_check(ast_model, r'C:\directory\file.robot', in_memory)
+        expected_issues = {
+            'Missing documentation in suite',
+            'Section is empty',            
+            'Too many blank lines at the end of file'
+        }
+        assert all(issue.desc in expected_issues for issue in issues)
+
+    def test_run_check_in_memory_with_windows_line_endings(self):
+        config = robocop.Config(root='.')
+
+        robocop_runner = robocop.Robocop(config=config)
+        robocop_runner.reload_config()
+        in_memory = "*** Settings *** \r\n\r\n"
+        ast_model = get_model(in_memory)
+        issues = robocop_runner.run_check(ast_model, r'C:\directory\file.robot', in_memory)
+        expected_issues = {
+            'Missing documentation in suite',
+            'Section is empty',
+            'Trailing whitespace at the end of line',
+            'Too many blank lines at the end of file'
+        }
+        assert all(issue.desc in expected_issues for issue in issues)
+
+    def test_run_check_in_memory_with_mac_line_endings(self):
+        config = robocop.Config(root='.')
+
+        robocop_runner = robocop.Robocop(config=config)
+        robocop_runner.reload_config()
+        in_memory = "*** Settings *** \r\r"
+        ast_model = get_model(in_memory)
+        issues = robocop_runner.run_check(ast_model, r'C:\directory\file.robot', in_memory)
         expected_issues = {
             'Missing documentation in suite',
             'Section is empty',
+            'Trailing whitespace at the end of line',
             'Too many blank lines at the end of file'
         }
         assert all(issue.desc in expected_issues for issue in issues)
```

Comment:
I think we need to reverse condition - right now if the issues is empty or only contains 1 issue it will be true (as long as given issue is inside expected issues). We actually rather want to check that "all the expected issues are in issues". That's why we missed 'Too many blank lines at the end of file' but the test passed.